rules/codelingo/go/reassigned-error/codelingo.yaml

```diff
@@ -14,8 +14,45 @@ funcs:
 tenets:
 - name: reassigned-error
   actions:
+      codelingo/docs:
+        title: Check `err` Before Variable Re-Assignment
+        body: | 
+          When re-assigning a variable in Go, it is convention to check the `err` prior to
+          doing so. First, check that the `err` is not equal to `nil` and then re-assign
+          the variable to avoid a panic.
+
+          ## Bad code
+          ```golang
+          \\ FileAppendOnly opens a file and appends an error code if the file is unable to be opened
+          func setFileAppendOnly(filename string) error {
+            file, err := os.Open("userdata.json")
+            defer file.Close()
+            err = file.Chmod(os.ModeAppend) // `err` not checked before re-assignment
+            if err != nil { 
+              return err
+            }
+            return nil
+          }
+          ```
+
+          ## Good code
+          ```golang
+          // FileAppendOnly opens a file and appends an error code if the file is unable to be opened
```

Comment:
```suggestion
          // FileAppendOnly sets a given file's mode to append only.
```

Reply:
pretty sure it doesn't do that. https://play.golang.org/p/Y0OuHpMGN-W